uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-balance-tab-filters.component.js

```diff
@@ -12,6 +12,7 @@ define([
       absencePeriods: '<',
       absenceTypes: '<',
       loggedInContactId: '<',
+      lookupContacts: '<',
```

Comment:
You need to initialize the contact_id filter in this file ([check this](https://github.com/civicrm/civihr/pull/2445/files#diff-7c168ab0b0a7a6aeb18c4f1f49395dffR28)) and [check if it's initialized correctly](https://github.com/civicrm/civihr/blob/d550a20ddb2a8437e0def2dfdc3c9881335568d7/uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-balance-tab-filters.component.spec.js#L46).

Reply:
The following discussion I had with @reneolivo which explains why it was done like this.

Debarshi Bhaumik: I initially did that, but there is another problem. That is the contact_id is not a mandatory filter. so initially it stays null, and because of that, "areFiltersReady()" never returns true initially. So the loader is shown. And once I select a contact, then only the results are shown. That is the reason its not added initially

René Olivo: let me check that function, 1 sec

Debarshi Bhaumik: I have thought about another solution, that is to initiate contact_id as undefined, instead of null. That way areFiltersReady() would work initially

Debarshi Bhaumik: yep sure

René Olivo: I see, leave it like that then, the other option would be to set it as undefined, but that's just confusing

Debarshi Bhaumik: you mean, no need to change anything? I also thought that not having it initially makes more sence than setting undefined or changing the areFiltersReady() function

René Olivo: yes, don't worry about that one, I don't have a better suggestion for now